receiver/oktareceiver/factory.go

```diff
@@ -0,0 +1,56 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      Okta://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oktareceiver
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+
+	"github.com/observiq/bindplane-agent/receiver/oktareceiver/internal/metadata"
+)
+
+// NewFactory creates a new factory for the Okta receiver
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		metadata.Type,
+		createDefaultConfig,
+		receiver.WithLogs(createLogsReceiver, metadata.LogsStability),
+	)
+}
+
+func createLogsReceiver(
+	_ context.Context,
+	params receiver.Settings,
+	rConf component.Config,
+	consumer consumer.Logs,
+) (receiver.Logs, error) {
+	cfg := rConf.(*Config)
+	r, err := newOktaLogsReceiver(cfg, params.Logger, consumer)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create an Okta log receiver instance: %w", err)
+	}
+	return r, nil
+}
+
+func createDefaultConfig() component.Config {
+	c := &Config{
+		PollInterval: defaultPollInterval,
```

Comment:
I'd set the default start time here:
```suggestion
        c := &Config{
            PollInterval: defaultPollInterval,
            StartTime: time.Now().UTC().Add(-defaultPollInterval)
        }
```

Then the relevant part of `newOktaLogsReceiver()` can be simplified to:
```
startTime, err := time.Parse(OktaTimeFormat, cfg.StartTime)
if err != nil {
    return nil, err
}
```